src/main/java/executor/command/CommandGetSpendingByMonth.java

```diff
@@ -26,16 +28,55 @@ public void execute(TaskList taskList) {
 
     @Override
     public void execute(Wallet wallet) {
-        ReceiptTracker receiptsInMonth = new ReceiptTracker();
-        String monthStr = Parser.parseForPrimaryInput(CommandType.EXPENDEDMONTH, userInput);
-        int month = monthStrToInt(monthStr);
-        if (month != 0) {
-            int year = Integer.parseInt(Parser.parseForFlag("year", userInput));
-            receiptsInMonth = wallet.getReceipts().findReceiptByMonthYear(month, year);
-            Double totalMoney = receiptsInMonth.getTotalCashSpent();
-            Ui.dukeSays("The total amount of money spent in " + monthStr + " " + year + " : " + totalMoney);
-        } else {
-            Ui.dukeSays("Invalid input, CORRECT FORMAT : expendedmonth <month> <year> ");
+        try {
+            ReceiptTracker receiptsInMonth = new ReceiptTracker();
+            String monthStr = (Parser.parseForPrimaryInput(CommandType.EXPENDEDMONTH, userInput)).toLowerCase();
+
+            if (monthStr.isEmpty()) {
+                Ui.dukeSays("No month input detected. FORMAT : expendedmonth <month> /year <year>");
+                return;
+            }
+
+            int month = monthStrToInt(monthStr);
+
+            if (month == 0) {
+                Ui.dukeSays("Wrong month input. Check Spelling");
+                return;
+            }
+
+            String yearStr = Parser.parseForFlag("year", userInput);
+
+            if (yearStr.isEmpty()) {
+                Ui.dukeSays("No year input detected. FORMAT : expendedmonth <month> /year <year>");
+                return;
+            }
+
+            if (yearStr.length() != 4) {
+                Ui.dukeSays("Year input contains extra number of variables. FORMAT : expendedmonth <month> /year <year>");
+                return;
+            }
+
+            try {
+                int year = Integer.parseInt(yearStr);
```

Comment:
try catch should be enclosed around this line. Under catch, print your message to the user and just call return to break.

Reply:
I have another overarching try-catch block above in line 31. They are present to catch different exceptions and to provide more specific error messages.